src/devices/Is31fl3730/Is31fl3730.cs

```diff
@@ -0,0 +1,381 @@
+
+           x63       x62        x61
+        ________  _________  ________
+        m2 | m1 || m2 | m1 || m2 | m1
+
+        This ordering makes sense if you are scrolling content right to left.
+
+        More detailed information about the structure of each pair follows (further down).
+
+        */
+
+        /// <summary>
+        /// Default I2C address for device.
+        /// </summary>
+        public const int DefaultI2cAddress = 0x61;
+
+        /// <summary>
+        /// Supported I2C addresses for device.
+        /// </summary>
+        public static readonly int[] SupportedI2cAddresses = new int[] { DefaultI2cAddress, 0x62, 0x63 };
+
+        /// <summary>
+        /// Brightness of LED matrix (override default value (128; max brightness)); set before calling Initialize method).
+        /// </summary>
+        public int Brightness = 0;
+
+        /// <summary>
+        /// Full current setting for each row output of LED matrix (override default value (40 mA)); set before calling Initialize method).
+        /// </summary>
+        public Current Current = 0;
+
+        /// <summary>
+        /// Matrix mode (override default value (8x8); set before calling Initialize method).
+        /// </summary>
+        public MatrixMode MatrixMode = 0;
+
+        /// <summary>
+        /// Display mode (override default value (Matrix 1 only); set before calling Initialize method).
+        /// </summary>
+        public DisplayMode DisplayMode = 0;
+
+        /// <summary>
+        /// Initialize LED driver.
+        /// </summary>
+        public void Initialize()
+        {
+            if (MatrixMode > 0)
+            {
+                _configurationValue |= (int)MatrixMode;
+            }
+
+            if (DisplayMode > 0)
+            {
+                _enabled[0] = DisplayMode is DisplayMode.MatrixOneOnly or DisplayMode.MatrixOneAndTwo;
+                _enabled[1] = DisplayMode is DisplayMode.MatrixTwoOnly or DisplayMode.MatrixOneAndTwo;
+                _configurationValue |= (int)DisplayMode;
+            }
+            else
+            {
+                _enabled[0] = true;
+            }
+
+            if (_configurationValue > 0)
+            {
+                Write(CONFIGURATION_REGISTER, (byte)_configurationValue);
+            }
+
+            if (Current > 0)
+            {
+                Write(LIGHTING_EFFECT_REGISTER, (byte)Current);
+            }
+
+            if (Brightness > 0)
+            {
+                Write(PWM_REGISTER, (byte)Brightness);
+            }
+        }
+
+        /// <summary>
+        /// Write LED for matrix.
+        /// </summary>
+        /// <param name="matrix">The matrix to use.</param>
+        /// <param name="x">The x dimension for the LED.</param>
+        /// <param name="y">The y dimension for the LED.</param>
+        /// <param name="value">The value to write.</param>
+        public void WriteLed(int matrix, int x, int y, int value)
```

Comment:
Something to consider as you only have 2 matrix: use an enum rather than an int?